siteshot.coffee

```diff
@@ -30,24 +30,34 @@ class SiteShot
               ph.createPage (page) =>
                 page.open route, (status) =>
                   if status is 'success'
-                    # Get page HTML
-                    page.evaluate (-> document.documentElement.outerHTML), (res) =>
-                      # Set snapshot name
-                      snapPrefix = if url.parse(route).path is '/'
-                        '/index'
-                      else
-                        url.parse(route).path
-                      snapPath = "#{config.snapshotDir}#{snapPrefix}.html"
-                      # Create directory
-                      mkdirp path.dirname(snapPath), (err) =>
-                        throw err if err?
+                    setTimeout (->
+                      generateHTML = ->
+                        # Get page HTML
+                        page.evaluate (-> document.documentElement.outerHTML), (res) =>
+                          # Set snapshot name
+                          snapPrefix = if url.parse(route).path is '/'
+                            '/index'
+                          else
+                            url.parse(route).path
+                          snapPath = "#{config.snapshotDir}#{snapPrefix}.html"
+                          # Create directory
+                          mkdirp path.dirname(snapPath), (err) =>
+                            throw err if err?
+
+                            # Write snapshot file
+                            fs.writeFile snapPath, res, (err) =>
+                              throw err if err?
+                              page.close()
+                              console.log "Finish loading #{route} and save it in #{snapPath}"
+                              callback()
 
-                        # Write snapshot file
-                        fs.writeFile snapPath, res, (err) =>
-                          throw err if err?
-                          page.close()
-                          console.log "Finish loading #{route} and save it in #{snapPath}"
-                          callback()
+                      if (typeof config.pageModifier == 'function')
+                        config.pageModifier(page, ->
+                          generateHTML();
+                        );
+                      else
+                        generateHTML();
+                    ), config.delay || 0
```

Comment:
`or` is preferred than `||` by [CoffeeScript style guide](https://github.com/polarmobile/coffeescript-style-guide#miscellaneous)